examples/webgl_materials_cubemap_mipmaps.html

```diff
@@ -0,0 +1,281 @@
+				gl_Position = projectionMatrix * mvPosition;
+
+			}
+
+		</script>
+
+		<script type="module">
+
+			import * as THREE from '../build/three.module.js';
+
+			import Stats from './jsm/libs/stats.module.js';
+
+			import { OrbitControls } from './jsm/controls/OrbitControls.js';
+
+			var container, stats;
+
+			var camera, scene, renderer, backgound;
+
+			init();
+			animate();
+
+			// load custmized cube texture
+			async function loadCubeTexture( path, imageWidth ) {
+
+				//fetch data
+				var response = await fetch( path );
+				var arrayBuffer = await response.arrayBuffer();
+
+				var cubeTextures = [];
+				var byteOffset = 0;
+				const maxLevel = Math.log( imageWidth ) / Math.LN2;
+
+				for ( var level = 0; level <= maxLevel; ++ level ) {
+
+					var size = Math.pow( 2, maxLevel - level );
+					var imageByteLength = size * size * 4;
+					var textures = [];
+					
+					if ( byteOffset >= arrayBuffer.byteLength ) {
+
+						break;
+
+					}
+
+					for ( var face = 0; face < 6; ++ face ) {
+
+						var imageData = new Uint8Array( arrayBuffer, byteOffset, imageByteLength );
+						var texture = new THREE.DataTexture( imageData, size, size, THREE.RGBAFormat, THREE.UnsignedByteType );
+						textures.push( texture );
+						byteOffset += imageByteLength;
+					}
+
+					var levelCubeTexture = new THREE.CubeTexture( textures );
+					cubeTextures.push( levelCubeTexture );
+				}
+
+				var customizedCubeTexture = cubeTextures.shift();
+				customizedCubeTexture.minFilter = THREE.LinearMipMapLinearFilter;
+				customizedCubeTexture.magFilter = THREE.LinearFilter;
+				customizedCubeTexture.format = THREE.RGBAFormat;
+
+				if ( cubeTextures.length > 0 ) {
+
+					customizedCubeTexture.mipmaps = cubeTextures;
+					customizedCubeTexture.generateMipmaps = false;
+
+				}
+
+				customizedCubeTexture.needsUpdate = true;
+
+				return customizedCubeTexture;
+
+			}
+
+			function init() {
+
+				container = document.createElement( 'div' );
+				document.body.appendChild( container );
+
+				camera = new THREE.PerspectiveCamera( 50, window.innerWidth / window.innerHeight, 1, 100000 );
+				camera.position.z = 4000;
+
+				//load cubemap
+
+				scene = new THREE.Scene();
+
+				//lights
+				var ambient = new THREE.AmbientLight( 0xffffff, 0.6 );
+				scene.add( ambient );
+
+				var pendings = [];
+				loadCubeTexture( 'textures/cube/angus/specular_cubemap_256_luv.bin', 256 ).then( function ( reflectCubeTexture ) {
+
+					//rewrite mipmaps
+					var originalCubeTexture = reflectCubeTexture.clone();
+					originalCubeTexture.generateMipmaps = true;
+
+					//models
+					var sphere = new THREE.SphereBufferGeometry( 100, 128, 128 );
+
+					//materials
+					var uniforms = {
+						lodRange: { value: new Float32Array( [ 5, 8 ] ) }, // min & max lod
+						roughness: { value: 0 },
+						specularCubeMap: { value: reflectCubeTexture }
+					};
+
+					var defines = {
+						LUV: 1
+					};
+
+					var material = new THREE.ShaderMaterial( {
```

Comment:
Why using a custom shader? Can you use a built-in material for this?

Reply:
Well, that would be better. But my customized cube map textures was saved in LogLuv color format which threejs couldn't decode in shader.

Reply:
Does it work if you set the `encoding` property of the texture to `THREE.LogLuvEncoding`?

Reply:
I have tried out. WebGL will raise `invalid format` warning

Like That:
![image](https://user-images.githubusercontent.com/19382361/61946524-64782100-afd5-11e9-8419-ad716d98deaf.png)

Reply:
Sorry, but I think it's important to fix this issue rather than adding custom shader code to the example.

Reply:
i have placed custom shader material with mesh phong material. And fix logluv encoding type error, Which seam to be a missing switch branch in `WebGLPrograms::getEncodingComponents` function

Reply:
Great work! 👍